src/hugr/validate.rs

```diff
@@ -228,7 +228,9 @@ impl<'a> ValidationContext<'a> {
         // Input ports and output linear ports must always be connected
         let mut links = self.hugr.graph.port_links(port_index).peekable();
         let must_be_connected = match dir {
-            Direction::Incoming => port_kind.is_linear() || matches!(port_kind, EdgeKind::Const(_)),
+            Direction::Incoming => {
```

Comment:
Whoa, hang on. Drive-by I realize, but shouldn't this be `Direction::Incoming => true` ?

Reply:
Nop, incoming order ports can be disconnected.

Reply:
Ah, but the check is still wrong as we are allowing more things besides order edges. I'll fix it.

Reply:
Oh, yes, we use ports for those in the implementation, but not the spec.

In which case let's update the comment two lines above; the word "ports" there must be interpreted as ports in the spec, not ports in the implementation.

The word we're looking for might be "dataflow" (at least the spec defines dataflow edges as value or static)